plugin/pkg/utils/cloudprofile.go

```diff
@@ -60,13 +62,70 @@ func ValidateCloudProfileChanges(namespacedCloudProfileLister gardencorev1beta1l
 	if err != nil {
 		return err
 	}
-	if oldCloudProfileReference.Kind == constants.CloudProfileReferenceKindCloudProfile &&
-		newCloudProfileReference.Kind == constants.CloudProfileReferenceKindNamespacedCloudProfile &&
-		equality.Semantic.DeepEqual(oldCloudProfileReference, newCloudProfileRoot) {
-		return nil
+	oldCloudProfileRoot, err := getRootCloudProfile(namespacedCloudProfileLister, oldCloudProfileReference, oldShoot.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if !equality.Semantic.DeepEqual(oldCloudProfileRoot, newCloudProfileRoot) {
+		fromProfile := fmt.Sprintf("%q", oldCloudProfileReference.Name)
+		if oldCloudProfileReference.Kind != constants.CloudProfileReferenceKindCloudProfile {
+			fromProfile += fmt.Sprintf(" (root: %q)", oldCloudProfileRoot.Name)
+		}
+		toProfile := fmt.Sprintf("%q", newCloudProfileReference.Name)
+		if newCloudProfileReference.Kind != constants.CloudProfileReferenceKindCloudProfile {
+			toProfile += fmt.Sprintf(" (root: %q)", newCloudProfileRoot.Name)
+		}
+		return fmt.Errorf("cloud profile reference change is invalid: cannot change from %s to %s. The cloud profile reference must remain within the same hierarchy", fromProfile, toProfile)
 	}
 
-	return fmt.Errorf("a Seed's CloudProfile may only be changed to a descendant NamespacedCloudProfile, other modifications are currently not supported")
+	if !equality.Semantic.DeepEqual(newCloudProfileReference, oldCloudProfileReference) {
+		newCloudProfileSpec, err := GetCloudProfileSpec(cloudProfileLister, namespacedCloudProfileLister, newShoot)
+		if err != nil {
+			return fmt.Errorf("could not find cloudProfileSpec from the shoot cloudProfile reference: %s", err.Error())
+		}
+		if oldCloudProfileReference.Kind != newCloudProfileReference.Kind || oldCloudProfileReference.Name != newCloudProfileReference.Name {
```

Comment:
Isn't this case already covered by `if !equality.Semantic.DeepEqual(newCloudProfileReference, oldCloudProfileReference)` above?